src/components/StatusUpdater/UpdateStatusForm.tsx

```diff
@@ -0,0 +1,174 @@
+import { useState } from 'react';
+import { Box, DialogContentText, FormControl, FormHelperText, Radio, RadioGroup, TextField, Typography, Select, MenuItem, Stack } from '../Material';
+import { useAppDispatch } from '@respond/lib/client/store';
+import { ActivityActions } from '@respond/lib/state';
+import { Activity, Participant, ResponderStatus } from '@respond/types/activity';
+import { UserInfo } from '@respond/types/userInfo';
+import { Control, Controller, FieldErrors, Resolver, ResolverResult, SubmitHandler, useForm } from 'react-hook-form';
+import { FormControlLabel } from '@mui/material';
+
+interface FormValues {
+  miles: number|'',
+  addMiles: number|'',
+}
+
+export function useFormLogic(
+  activity: Activity,
+  user: UserInfo,
+  respondingOrgId: string,
+  participant: Participant|undefined,
+  currentStatus: ResponderStatus|undefined,
+  newStatus: ResponderStatus,
+  onFinish: () => void,
+) {
+  const dispatch = useAppDispatch();
+
+  const resolver: Resolver<FormValues> = async (values) => {
+    const result: ResolverResult<FormValues> = {
+      values: values ? values : {},
+      errors: {},
+    };
+
+    if (values.addMiles != null && values.addMiles < 0) {
+      result.errors.addMiles = { type: 'min', message: 'Must be a positive number' };
+    }
+    if (values.miles != null && values.miles < 0) {
+      result.errors.miles = { type: 'min', message: 'Must be a positive number' };
+    }
+
+    return result;
+  }
+
+  const form = useForm<FormValues>({
+    resolver,
+    defaultValues: { miles: participant?.miles ?? '', addMiles: '' },
+  });
+  if (form.getValues().miles !== (participant?.miles ?? '')) {
+    form.reset({ miles: participant?.miles ?? '', addMiles: '' });
+  }
+
+
+  const onSubmit: SubmitHandler<FormValues> = data => {
+
+    if (data.addMiles !== undefined) {
+      data.miles = Number(data.miles ?? 0) + Number(data.addMiles);
+    }
+
+    dispatch(ActivityActions.participantUpdate(
+      activity.id,
+      user.participantId,
+      user.given_name ?? '',
+      user.family_name ?? '',
+      respondingOrgId,
+      new Date().getTime(),
+      newStatus,
+      data.miles === '' ? undefined : data.miles,
+    ));
+    onFinish();
+  };
+
+  return {
+    ...form,
+    errors: form.formState.errors,
+    doSubmit: form.handleSubmit(onSubmit),
+    context: {
+      newStatus,
+      activity,
+      participant,
+    },
+  };
+}
+type FormType = ReturnType<typeof useFormLogic>;

```

Comment:
Maybe it's just me, but I found "FormType" to be a pretty confusing name. That made it sound like it's an enum or specialization of a generic form class (e.g. StatusForm, ContactForm, etc).

Best alternative that comes to mind is "FormLogic", tying into the name of the function. It's also fairly descriptive of what the type is.